module-2/backend/week_one.md

```diff
@@ -5,30 +5,80 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. List the five common HTTP verbs and what the purpose is of each verb.
-2. What is Sinatra?
+  POST - create
+  GET - read
+  PUT - update/replace
+  PATCH - update/modify
+  DELETE - delete
+  
+2. What is Sinatra? 
+  a framework for Ruby
+
 4. What is MVC?
+  the common folder structure for web applications consisting of model-the info tracker/where 'work' takes place, views-what users see, controller-the delegater/sends you to the correct location
+  
 5. Why do we follow conventions when creating our actions/path names in our Sinatra routes?
+  to make our code easy to read/follow for other developers
+  
 6. What types of variables are accessible in our view templates without explicitly passing them?
+  HTML elements; all variables/data are sent from the controller
+  
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
   
   ```ruby
   get '/horses' do
+    @count = 1
+    @name = Mr. Ed
     erb :index
   end
   ```
 
 8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
+  ^see above^
+  
 9. What's the purpose of ERB?
+  Embedded Ruby files, enables you to embed Ruby code/methods into HTML output for web pages
+  
 10. Why do I need a development AND test database?
+  because they set the environment (development, test, or production)
+  
 11. What's responsive design?
+  the approach to building web pages/apps to fit multiple screen sizes
+
 12. What is CRUD and why is it important?
+  Create, Read, Update, Delete; basic necessary functions for any interactive application
+  
 13. What does HTTP stand for? 
+  Hyper Text Transfer Protocol
+  
 14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+  <%= renders ruby code on screen for user to see output %> <% runs ruby code behind the scenes %> 
+
 15. What's an ORM?
+   Object Relational Mappers - they talk between the database and Ruby
+   
 16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
+  ActiveRecord
+  
 17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+  GET 'restaurants/index' - reads/renders path for index page
+  GET 'restaurants/new' - reads data that's input for a new restaurant
+  POST 'resaurants/:id' -  writes the data that was input into new resaurant/id 

```

Comment:
POST doesn't take an `/:id` at the end. It helps me to remember that at this point the restaurant doesn't *have* an id since it hasn't been persisted to the database yet.